cryptosuite/bbsplussignatureproofsuite.go

```diff
@@ -0,0 +1,452 @@
+package cryptosuite
+
+import (
+	gocrypto "crypto"
+	"encoding/base64"
+	"strings"
+
+	"github.com/TBD54566975/ssi-sdk/crypto"
+	. "github.com/TBD54566975/ssi-sdk/util"
+	"github.com/goccy/go-json"
+	"github.com/pkg/errors"
+)
+
+const (
+	BBSPlusSignatureProof2020 SignatureType = "BbsBlsSignatureProof2020"
+)
+
+type BBSPlusSignatureProofSuite struct {
+	CryptoSuiteProofType
+}
+
+func GetBBSPlusSignatureProofSuite() *BBSPlusSignatureProofSuite {
+	return new(BBSPlusSignatureProofSuite)
+}
+
+// CryptoSuiteInfo interface
+
+var _ CryptoSuiteInfo = (*BBSPlusSignatureProofSuite)(nil)
+
+func (BBSPlusSignatureProofSuite) ID() string {
+	return BBSPlusSignatureSuiteID
+}
+
+func (BBSPlusSignatureProofSuite) Type() LDKeyType {
+	return BBSPlusSignatureSuiteType
+}
+
+func (BBSPlusSignatureProofSuite) CanonicalizationAlgorithm() string {
+	return BBSPlusSignatureSuiteCanonicalizationAlgorithm
+}
+
+func (BBSPlusSignatureProofSuite) MessageDigestAlgorithm() gocrypto.Hash {
+	return BBSPlusSignatureSuiteDigestAlgorithm
+}
+
+func (BBSPlusSignatureProofSuite) SignatureAlgorithm() SignatureType {
+	return BBSPlusSignatureProof2020
+}
+
+func (BBSPlusSignatureProofSuite) RequiredContexts() []string {
+	return []string{BBSSecurityContext}
+}
+
+// SelectivelyDisclose takes in a credential and  a map of fields to disclose as an LD frame
```

Comment:
This godoc needs some love. I've tried my best. 
```suggestion
// SelectivelyDisclose takes in a credential (parameter `p` that's Provable) and a map of fields to disclose as an LD frame, and produces a map of the JSON representation of the derived credential. The derived credential only contains the information that was specified in the LD frame, and a proof that's derived from the original credential. Note that a requirement for `p` is that the property `"proof"` must be present when it's marshaled to JSON, and it's value MUST be an object that conforms to a `BBSPlusProof`.
```